core/src/main/java/io/kestra/core/validations/validator/FlowValidator.java

```diff
@@ -92,6 +96,50 @@ public boolean isValid(
             .filter(label -> label.key() != null && label.key().startsWith(SYSTEM_PREFIX) && !label.key().equals(READ_ONLY))
             .forEach(label -> violations.add("System labels can only be set by Kestra itself, offending label: " + label.key() + "=" + label.value()));
 
+        List<Pattern> inputsWithMinusPatterns = Optional.ofNullable(value.getInputs())
+            .orElse(new ArrayList<>())
+            .stream()
+            .filter(input -> input.getId().contains("-"))
+            .map(input -> Pattern.compile("\\{\\{\\s*inputs." + input.getId().toString() + "\\s*\\}\\}"))
```

Comment:
```suggestion
            .map(input -> Pattern.compile("\\{\\{\\s*inputs." + input.getId() + "\\s*\\}\\}"))
```

Input.getId() is already a String